source/extensions/filters/listener/http_inspector/http_inspector.cc

```diff
@@ -32,88 +31,38 @@ http_parser_settings Filter::settings_{
     nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr,
 };
 
-Network::FilterStatus Filter::onAccept(Network::ListenerFilterCallbacks& cb) {
-  ENVOY_LOG(debug, "http inspector: new connection accepted");
-
-  const Network::ConnectionSocket& socket = cb.socket();
-
-  const absl::string_view transport_protocol = socket.detectedTransportProtocol();
-  if (!transport_protocol.empty() && transport_protocol != "raw_buffer") {
-    ENVOY_LOG(trace, "http inspector: cannot inspect http protocol with transport socket {}",
-              transport_protocol);
-    return Network::FilterStatus::Continue;
-  }
-
-  cb_ = &cb;
-  const ParseState parse_state = onRead();
+Network::FilterStatus Filter::onData(Network::ListenerFilterBuffer& buffer) {
+  auto raw_slice = buffer.rawSlice();
+  const char* buf = static_cast<const char*>(raw_slice.mem_);
+  const auto parse_state = parseHttpHeader(absl::string_view(buf, raw_slice.len_));
   switch (parse_state) {
   case ParseState::Error:
-    // As per discussion in https://github.com/envoyproxy/envoy/issues/7864
-    // we don't add new enum in FilterStatus so we have to signal the caller
-    // the new condition.
-    cb.socket().close();
-    return Network::FilterStatus::StopIteration;
+    // Invalid HTTP preface found, then just continue for next filter.
+    done(false);
+    return Network::FilterStatus::Continue;
```

Comment:
This is where I'm wrong.

The original code is about the socket reading fail, then it close the socket and stop the iteration, waiting for close.
But with this PR, the `peek` was done in the core code, not in each filter anymore, so the filter needn't to care about this case.

The `ParseState::Error` here is about parse invalid HTTP preface. For invalid http preface, we won't close the socket, just continue to next filter.  This actually match the code at line 69.

Reply:
I tested locally by changing this line and it passed Istio CI. So if this is all that's needed, I think it should be OK now.

Reply:
For my understanding - does `Continue` mean that this listener filter is no longer called back, and we move on to the next listener filter?

Reply:
Yes, `Continue` means moving on. `StopIteration` means waiting for more data.   Close socket + `StopIteration` means end the connection.